ambry-protocol/src/main/java/com/github/ambry/protocol/PurgeRequest.java

```diff
@@ -0,0 +1,129 @@
+ */
+public class PurgeRequest extends RequestOrResponse {
+
+  static final short PURGE_MESSAGE_REQUEST_VERSION_1 = 1;
+  private final static short CURRENT_VERSION = PURGE_MESSAGE_REQUEST_VERSION_1;
+
+  private final BlobId blobId;
+
+  /**
+   * Helper to construct PurgeRequest from a stream
+   * @param stream the stream to read data from
+   * @param map the {@link ClusterMap} to use
+   * @return a TtlUpdateRequest based on data read off of the stream
+   * @throws IOException if there were any problems reading the stream
+   */
+  public static PurgeRequest readFrom(DataInputStream stream, ClusterMap map) throws IOException {
+    short version = stream.readShort();
+    if (version == PURGE_MESSAGE_REQUEST_VERSION_1) {
+      return CompactMessageRequest_V1.readFrom(stream, map);
+    }
+    throw new IllegalStateException("Unknown Purge Request version " + version);
+  }
+
+  /**
+   * @param correlationId the correlation id for the request
+   * @param clientId the id of the client generating the request
+   * @param blobId the blob ID whose TTL needs to be updated
+   */
+  public PurgeRequest(int correlationId, String clientId, BlobId blobId) {
+    this(correlationId, clientId, blobId, CURRENT_VERSION);
+  }
+
+  /**
+   * @param correlationId the correlation id for the request
+   * @param clientId the id of the client generating the request
+   * @param blobId the blob ID that needs to be purged.
+   * @param version the version of the {@link PurgeRequest}.
+   */
+  PurgeRequest(int correlationId, String clientId, BlobId blobId, short version) {
+    super(PurgeRequest, version, correlationId, clientId);
+    this.blobId = blobId;
+  }
+
+  @Override
+  public void accept(RequestVisitor visitor) {
+    // TODO Efficient Metadata Operations: Implement this.
+  }
+
+  @Override
+  protected void prepareBuffer() {
+    super.prepareBuffer();
+    bufferToSend.writeBytes(blobId.toBytes());
+  }
+
+  /**
+   * @return the blob ID that needs to be purged.
+   */
+  public BlobId getBlobId() {
+    return blobId;
+  }
+
+  /**
+   * @return the id of the account that the blob belongs to
+   */
+  public short getAccountId() {
+    return blobId.getAccountId();
+  }
+
+  /**
+   * @return the id of the container that the blob belongs to
+   */
+  public short getContainerId() {
+    return blobId.getContainerId();
+  }
+
+  @Override
+  public long sizeInBytes() {
+    // header + blobId
+    return super.sizeInBytes() + blobId.sizeInBytes();
+  }
+
+  @Override
+  public String toString() {
+    return "PurgeRequest[" + "BlobID=" + blobId + ", " + "PartitionId=" + blobId.getPartition() + ", " + "ClientId="
+        + clientId + ", " + "CorrelationId=" + correlationId + ", " + "AccountId=" + blobId.getAccountId() + ", "
+        + "ContainerId=" + blobId.getContainerId() + "]";
+  }
+
+  /**
+   * Class to read protocol version 1 PurgeRequest from the stream.
+   */
+  public static class CompactMessageRequest_V1 {
```

Comment:
Shall we use consistent name, both Purge or both Compact?
I prefer Purge more, since we have used compact for server already.

Reply:
fixed it.